File: example-java/app/src/main/java/com/vibes/androidsdkexampleapp/api/VibesAPI.java
package com.vibes.androidsdkexampleapp.api;

import android.content.Context;
import android.util.Log;

import com.vibes.androidsdkexampleapp.notification.MyNotificationFactory;
import com.vibes.vibes.Credential;
import com.vibes.vibes.Vibes;
import com.vibes.vibes.VibesListener;

/**
 * Created by jean-michel.barbieri on 2/25/18
 * Copyright (c) Vibes 2018 . All rights reserved.
 * Last modified 12:33 AM
 */
public class VibesAPI implements VibesAPIContract {

    public VibesAPI(Context context){
        Vibes.getInstance().setNotificationFactory(new MyNotificationFactory(context));
    }

    @Override
    public void registerDevice(VibesListener<Credential> listener) {
        Vibes.getInstance().registerDevice(listener);
    }

    @Override
    public void unregisterDevice(VibesListener<Void> listener) {
        Vibes.getInstance().unregisterDevice(listener);
    }

    @Override
    public void registerPush(VibesListener<Void> listener, String firebasePushToken) {
        Log.d("Controller", "--> Controller Firebase Token: " + firebasePushToken);
        Vibes.getInstance().registerPush(firebasePushToken, listener);
    }

    @Override
    public void unregisterPush(VibesListener<Void> listener) {
        Vibes.getInstance().unregisterPush(listener);
    }
}

File: example-kotlin/app/src/main/java/com/android/exampleapp/kotlin/api/VibesAPI.kt
package com.android.exampleapp.kotlin.api

import android.content.Context
import android.util.Log
import com.android.exampleapp.kotlin.notification.MyNotificationFactory
import com.vibes.vibes.Credential
import com.vibes.vibes.Vibes
import com.vibes.vibes.VibesListener

class VibesAPI: VibesAPIContract {

    constructor(context: Context){
        Vibes.getInstance().setNotificationFactory(MyNotificationFactory(context))
    }

    override fun registerDevice(listener: VibesListener<Credential>) {
        Vibes.getInstance().registerDevice(listener)
    }

    override fun unregisterDevice(listener: VibesListener<Void>) {
        Vibes.getInstance().unregisterDevice(listener)
    }

    override fun registerPush(listener: VibesListener<Void>, firebasePushToken: String) {
        Log.d("Controller", "--> Controller Firebase Token: $firebasePushToken")
        Vibes.getInstance().registerPush(firebasePushToken, listener)
    }

    override fun unregisterPush(listener: VibesListener<Void>) {
        Vibes.getInstance().unregisterPush(listener)
    }
}